spacesuit/src/value.rs

```diff
@@ -107,6 +107,10 @@ impl SignedInteger {
             Some(self.0 as u64)
         }
     }
+
+    pub fn to_scalar(self) -> Scalar {
+        <SignedInteger as Into<Scalar>>::into(self)
```

Comment:
instead of defining this in terms of Into, lets do the reverse: put the `match{...}` here and call `to_scalar` inside impl of `Into<Scalar>`.